Docs: updated codeowners file

lwandz13 opened the pull request:
Under Technical documentation, I removed myself from data sources, and added  myself to admin (IAM)

lwandz13 committed:
removed myself from data sources, added to admin (IAM)

baldm0mma commented:
Looks like it's mad about the `blem` in the dockerfile?
```
Stdout:

Stderr:
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 5.80kB done
#1 DONE 0.0s
Dockerfile:1
--------------------
   1 | >>> blem# syntax=docker/dockerfile:1
   2 |     
   3 |     ARG BASE_IMAGE=alpine:3.18.5
--------------------
ERROR: failed to solve: dockerfile parse error on line 1: unknown instruction: blem#
```

briangann commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-# syntax=docker/dockerfile:1
+blem# syntax=docker/dockerfile:1
```
this appears to be unintended extra text

lwandz13 replied:
this has been removed - thank you for catching it!

jdbaldry committed:
Remove accidental changes

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Rewrite Technical documentation codeowners for readability and consistency

Now it is consistent with the comments at the head of the file, at
least for this section of the repo.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry reviewed (approved):
Thanks Larissa, I've removed the accidental changes and reformatted the section but kept your CODEOWNERs change from datasources to administration.

baldm0mma commented:
@lwandz13 looks like the `continuous-integration/drone/pr` check got stuck on a flaky loki test. I restarted the drone run and it passed! I think you're good to go to merge. :)
